src/App.tsx

```diff
@@ -38,6 +39,16 @@ const getPersistentTokens = (): OptionalAuthTokens => {
 };
 
 export default function App(): JSX.Element {
+  const hasReferrer = !!document.referrer;
+
+  const searchParams: SearchParameters = urlUtils.parseSearchParameters(
+    window.location.search
```

Comment:
Minor comment: we could add window check to avoid crash if in the future hauki goes SSR